crypto/s2n_pkey.c

```diff
@@ -184,7 +184,10 @@ int s2n_asn1der_to_public_key_and_type(struct s2n_pkey *pub_key, s2n_pkey_type *
 
     /* If cert parsing is successful, d2i_X509 increments *cert_to_parse to the byte following the parsed data */
     uint32_t parsed_len = cert_to_parse - asn1der->data;
-    S2N_ERROR_IF(parsed_len != asn1der->size, S2N_ERR_DECODE_CERTIFICATE);
+
+    /* Length check and allow for up to 1 trailing unparsed byte */
+    uint32_t trailing_bytes = asn1der->size - parsed_len;
+    S2N_ERROR_IF(trailing_bytes != 0 && trailing_bytes != 1, S2N_ERR_DECODE_CERTIFICATE);
```

Comment:
Using `ENSURE_POSIX` makes this a bit easier to read since it's making an assertion, rather than a refutation.

```suggestion
    ENSURE_POSIX(trailing_bytes <= 1, S2N_ERR_DECODE_CERTIFICATE);
```

Reply:
thanks, it's updated!